src/components/video-card/VideoCard.jsx

```diff
@@ -2,46 +2,90 @@ import {
   BsThreeDotsVertical,
   MdWatchLater,
   MdDelete,
+  MdPlaylistAdd,
 } from "assets/icons/icons";
+import { useState } from "react";
 import "./video-card.css";
 
 const VideoCard = ({
-  cardData: { title, image, watchLater, history },
+  cardData,
   btnNameOne,
   addToHistory,
   removeFromHistory,
   addToWatchLater,
   removeFromWatchLater,
+  playlistDispatch,
 }) => {
+  const { title, image, watchLater, history } = cardData;
+  const [dropdown, setDropdown] = useState(false);
+
   return (
     <>
       <div className="video-card-container card-pos-rel">
-        {history ? (
+        {history && (
           <MdDelete
             className="video-card-icon cursor color-red"
             onClick={removeFromHistory}
           />
-        ) : (
-          <MdWatchLater
-            className={`video-card-icon cursor ${
-              watchLater ? "color-green" : "color-black"
-            }`}
-            onClick={watchLater ? removeFromWatchLater : addToWatchLater}
-          />
         )}
         <div className="video-card-image-container">
           <img className="image-resp" src={image} alt={title} />
         </div>
         <div className="card-margin">
           <div className="flex video-card-flex-adjustment">
             <h5>{title}</h5>
-            <BsThreeDotsVertical className="cursor" />
+            {!history && (
+              <BsThreeDotsVertical
+                className="cursor"
+                onClick={() => {
+                  setDropdown(!dropdown);
+                  playlistDispatch({ type: "VIDEO_DATA", payload: cardData });
+                }}
+              />
+            )}
           </div>
           <small className="text-gray">6k views | 4 hours ago</small>
         </div>
-        <button className="btn btn-primary m-b-4px" onClick={addToHistory}>
+        <button className="btn btn-primary" onClick={addToHistory}>
           {btnNameOne}
         </button>
+        {dropdown && (
+          <div className="dropdown">
+            <ul className="ul-none dropdown-ul">
+              <li
+                className="dropdown-li cursor"
+                onClick={
+                  watchLater
+                    ? () => {
+                        removeFromWatchLater();
+                        setDropdown(!dropdown);
+                      }
+                    : () => {
+                        addToWatchLater();
+                        setDropdown(!dropdown);
+                      }
```

Comment:
Suggestion: can extract function from here to another function and use , use it here